### ISSUE TITLE
t/parent-pmc.t still fails on non PMC perls with Config without non_bincompat_options

### ISSUE THREAD

**Issue by maxatome** [Corion/parent#4]
Hi,

As I commented on the last commit 4bf8adca6a55cae6cfd5842f224b1c0099375204 related to this bug, the correction still does not work for perl without PMC AND Config modules that does not implement `non_bincompat_options` method...

We encounter the case with perl-5.10.1 compiled without PMC support.

Note that, still in our case, `perl -V` does not list `PERL_DISABLE_PMC` in _Compile-time options_.

So one should add a else clause to the `if (Config->can('non_bincompat_options'))` as below:

``` perl
if (Config->can('non_bincompat_options')) {
    ...
} else {
    # Test directly in options used for the compilation
    $no_pmc = $Config{ccflags} =~ /-DPERL_DISABLE_PMC\b/
        || $Config{cppflags} =~ /-DPERL_DISABLE_PMC\b/;
}
```

**Comment by maxatome** [comment]
Still broken in 0.233:

```
t/parent-pmc.t ................. 1/3 
#   Failed test 'Block ran until the end'
#   at t/parent-pmc.t line 41.
#          got: undef
#     expected: '1'

#   Failed test 'No error'
#   at t/parent-pmc.t line 42.
#          got: 'Can't locate FileThatOnlyExistsAsPMC.pm in @INC (@INC contains: t/lib ... .) at .../parent-0.233/blib/lib/parent.pm line 16.
# BEGIN failed--compilation aborted at (eval 4) line 4.
# '
#     expected: ''

#   Failed test 'FileThatOnlyExistsAsPMC->can('exclaim')'
#   at t/parent-pmc.t line 45.
#     FileThatOnlyExistsAsPMC->can('exclaim') failed
# Looks like you failed 3 tests of 3.
t/parent-pmc.t ................. Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/3 subtests 
```

The proposed modification in the previous comment still corrects the problem...

**Comment by dolmen** [comment]
@corion I had proposed a fix in #3 that has been ignored and closed without a proper review.
Please review it again.

**Comment by dolmen** [comment]
Simpler fix is in #5

**Comment by maxatome** [comment]
0.234 corrects the problem. Thanks.
